annotation-processor/src/main/java/org/hibernate/validator/ap/classchecks/AbstractMethodOverrideCheck.java

```diff
@@ -0,0 +1,338 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.ap.classchecks;
+
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.Name;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.type.TypeMirror;
+import javax.lang.model.util.Elements;
+import javax.lang.model.util.Types;
+
+import org.hibernate.validator.ap.checks.ConstraintCheckIssue;
+import org.hibernate.validator.ap.util.CollectionHelper;
+
+/**
+ * Abstract base class for {@link ClassCheck} implementations that check overridden methods.
+ *
+ * @author Marko Bekhta
+ */
+public abstract class AbstractMethodOverrideCheck extends AbstractClassCheck {
+
+	private static final String JAVA_LANG_OBJECT = "java.lang.Object";
+
+	private static final String JAVAX_VALIDATION_CONSTRAINT = "javax.validation.Constraint";
+
+	private final Elements elementUtils;
+
+	private final Types typeUtils;
+
+	public AbstractMethodOverrideCheck(Elements elementUtils, Types typeUtils) {
+		this.elementUtils = elementUtils;
+		this.typeUtils = typeUtils;
+	}
+
+	@Override
+	public Collection<ConstraintCheckIssue> checkMethod(ExecutableElement currentMethod) {
```

Comment:
I'm not sure this can be implemented in the abstract method. I suspect the logic will be totally different between the parameters and the return values.